src/lib/MrDocsCompilationDatabase.cpp

```diff
@@ -411,17 +337,21 @@ adjustCommandLine(
         isExplicitCCompileCommand || (!isExplicitCppCompileCommand && isImplicitCSourceFile);
 
     constexpr auto is_std_option = [](std::string_view const opt) {
-        return opt.starts_with("-std=") || opt.starts_with("--std=") || opt.starts_with("/std:");
+        return opt.starts_with("-std=") || opt.starts_with("--std=")
+               || // clang options
+               opt.starts_with("-std:")
+               || opt.starts_with("/std:"); // clang-cl options
     };
-    if (std::ranges::find_if(cmdline, is_std_option) == cmdline.end())
+    if (std::ranges::none_of(cmdline, is_std_option))
     {
         if (!isCCompileCommand)
         {
-            new_cmdline.emplace_back("-std=c++23");
+            new_cmdline.emplace_back(
+                is_clang_cl ? "-std:c++23preview" : "-std=c++23");
         }
         else
         {
-            new_cmdline.emplace_back("-std=c23");
+            new_cmdline.emplace_back(is_clang_cl ? "-std:c17" : "-std=c23");
```

Comment:
Do they mean the same thing? Maybe, if that's possible, this case and the case above would be a good case for using that flag that bypasses the clang-cl conversions? Because I'm not sure someone else would understand exactly why these mean the same thing. In other words, it looks like code someone will want to change in passing.

Reply:
Changed to use the latest standard for both.